### ISSUE TITLE
[iOS][a11y]: Unable to run Audit with Accessibility Inspector

### ISSUE THREAD

**Issue by niteshsh4rma** [flutter/flutter#143146]
### Steps to reproduce

1. Run App either from VSCode or XCode.
2. Run Accessibility Inspector as mentioned in Flutter's Accessibility Doc.
3. Go to mid tab.

### Actual results

1. Unable to select target app in Simulator option because it does not provide any.
2. Audit option not available.

<img width="1195" alt="Screenshot 2024-02-08 at 12 48 39 PM" src="https://github.com/flutter/flutter/assets/58659088/c4f4b476-95f9-4f7a-bbab-31e1fcd690ae">

### Logs

<details open>
<summary>Logs</summary>

```console

<!-- Paste your logs here -->

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

[✓] Flutter (Channel master, 3.20.0-4.0.pre.2, on macOS 14.2 23C64 darwin-x64, locale en-IN)
    • Flutter version 3.20.0-4.0.pre.2 on channel master at /Users/admin/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fee1868d89 (25 hours ago), 2024-02-06 22:41:25 -0800
    • Engine revision 808886312e
    • Dart version 3.4.0 (build 3.4.0-104.0.dev)
    • DevTools version 2.32.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/admin/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.86.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.82.0

[✓] Connected device (3 available)
    • iPhone 15 Pro (mobile) • 4F17CAF2-4F56-4163-B89E-DC71AA0877C3 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 14.2 23C64 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 121.0.6167.139

[✓] Network resources
    • All expected network resources are available.

• No issues found!


```

</details>

**Comment by dam-ease** [comment]
Hello @niteshsh4rma.
Thanks for filing this. 
Following the steps you shared, I'm able to select a simulator and run Audit on my end.

<img width="930" alt="Screenshot 2024-02-08 at 12 48 41" src="https://github.com/flutter/flutter/assets/53122008/35572a79-eb76-473e-a4af-c1f5da10b22b">

Are there any extra steps I might be missing out on?

**Comment by niteshsh4rma** [comment]
@dam-ease Can you please share a video of doing this?

**Comment by dam-ease** [comment]
Hello @niteshsh4rma.
Here is a screen recording from my end.


https://github.com/flutter/flutter/assets/53122008/3266c189-989d-4517-8132-c0b99674aacb
